src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```
@@ -0,0 +1,45 @@
+package mate.academy.service.impl;
+
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        if (password == null || password.isEmpty()) {
+            throw new AuthenticationException("Password can't be empty");
+        }
+        User user = userService.findByEmail(email).orElseThrow(() ->
+                new AuthenticationException("User with email " + email + " not found"));
+        if (user.checkPassword(password)) {
+            return user;
+        } else {
+            throw new AuthenticationException("Wrong Password");
+        }
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        if (userService.findByEmail(email).isPresent()) {
+            throw new RegistrationException("User with email " + email + " already exist");

```

In register(), you call `userService.findByEmail(email).isPresent()` before validating `email`. If `email` is null or empty this can lead to unclear behavior coming from deeper layers. Consider validating `email` first and throwing a `RegistrationException` with a clear message (e.g. `"Email can't be empty"`) before calling userService. This will fail fast and provide clearer diagnostics.